src/batched/KokkosBatched_ApplyQ_Serial_Impl.hpp

```diff
@@ -0,0 +1,80 @@
+#ifndef __KOKKOSBATCHED_APPLY_Q_SERIAL_IMPL_HPP__
+#define __KOKKOSBATCHED_APPLY_Q_SERIAL_IMPL_HPP__
+
+
+/// \author Kyungjoo Kim (kyukim@sandia.gov)
+
+#include "KokkosBatched_Util.hpp"
+#include "KokkosBatched_ApplyQ_Serial_Internal.hpp"
+
+namespace KokkosBatched {
+
+  ///
+  /// Serial Impl
+  /// ===========
+
+  template<>
+  template<typename AViewType,
+           typename tViewType,
+           typename BViewType,
+           typename wViewType>
+  KOKKOS_INLINE_FUNCTION
+  int
+  SerialApplyQ<Side::Left,Trans::NoTranspose,Algo::ApplyQ::Unblocked>::
+  invoke(const AViewType &A,
+         const tViewType &t,
+         const BViewType &B,
+         const wViewType &w) {
+    return SerialApplyQ_LeftForwardInternal::
+      invoke(B.extent(0), B.extent(1), A.extent(1), 
+             A.data(), A.stride_0(), A.stride_1(),
+             t.data(), t.stride_0(), 
+             B.data(), B.stride_0(), B.stride_1(),
+             w.data());
+  }
+
+  template<>
```

Comment:
Why is there the empty `template<>` followed by another `template<...>`?

Reply:
This struct is specialized by its struct template arguments Side::Left, Trans::Trnaspose, Algo::ApplyQ::Unblocked. That is why it is empty but the struct has specialization while the function still can get views as a free type.